cni/network/network.go

```diff
@@ -151,32 +151,34 @@ func (plugin *netPlugin) Add(args *cniSkel.CmdArgs) error {
 		args.ContainerID, args.Netns, args.IfName, args.Args, args.Path)
 
 	defer func() {
-		if result == nil {
-			result = &cniTypesCurr.Result{}
-		}
-		
-		// Add Interfaces to result.
-		if result == nil {
-			result = &cniTypesCurr.Result{}
-		}
+		if err == nil {
+			if result == nil {
+				result = &cniTypesCurr.Result{}
+			}
 
-		iface = &cniTypesCurr.Interface{
-			Name: args.IfName,
-		}
+			// Add Interfaces to result.
+			if result == nil {
+				result = &cniTypesCurr.Result{}
+			}
 
-		result.Interfaces = append(result.Interfaces, iface)
+			iface = &cniTypesCurr.Interface{
+				Name: args.IfName,
+			}
 
-		addSnatInterface(nwCfg, result)
+			result.Interfaces = append(result.Interfaces, iface)
 
-		// Convert result to the requested CNI version.
-		res, err := result.GetAsVersion(nwCfg.CNIVersion)
-		if err != nil {
-			err = plugin.Error(err)
-		}
+			addSnatInterface(nwCfg, result)
 
-		// Output the result to stdout.
-		res.Print()
-		log.Printf("[cni-net] ADD command completed with result:%+v err:%v.", result, err)
+			// Convert result to the requested CNI version.
+			res, err := result.GetAsVersion(nwCfg.CNIVersion)
+			if err != nil {
+				err = plugin.Error(err)
+			}
+
+			// Output the result to stdout.
+			res.Print()
+			log.Printf("[cni-net] ADD command completed with result:%+v err:%v.", result, err)
```

Comment:
We should still write to the log file what happened to the CNI ADD call

Reply:
ok